azurerm/resource_arm_api_management_authorization_server.go

```diff
@@ -0,0 +1,492 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2018-01-01/apimanagement"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmApiManagementAuthorizationServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApiManagementAuthorizationServerCreateUpdate,
+		Read:   resourceArmApiManagementAuthorizationServerRead,
+		Update: resourceArmApiManagementAuthorizationServerCreateUpdate,
+		Delete: resourceArmApiManagementAuthorizationServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": azure.SchemaApiManagementChildName(),
+
+			"api_management_name": azure.SchemaApiManagementName(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"authorization_endpoint": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"client_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
```

Comment:
Could we validate this as a resource id:

Reply:
this is a user specifyable value (which can be anything) so unfortunately this is the best we can do here